### PR TITLE
chore(derive): data source unit tests

### LINKED ISSUE #162: chore(derive): Ethereum Data Source Tests
**Description**

> [!Warning]
> This is blocked by #159 

Add unit tests for the [EthereumDataSource](https://github.com/ethereum-optimism/kona/blob/main/crates/derive/src/sources/ethereum.rs).

These tests will need to use mock/test chain providers and blob providers for constructing the data source.

They should validate when `open_data` is called that
- if the `ecotone_timestamp` is **not** set and a calldata source should be returned.
- if the `ecotone_timestamp` **is** set and the `block_ref` timestamp is prior to the `ecotone_timestamp`, a calldata source is created.
- if the `ecotone_timestamp` **is** set and the `block_ref` timestamp is greater than or equal to the `ecotone_timestamp`, a blob provider is created.

### PR DESCRIPTION
**Pull request by merklefruit**
**Description**

- adds unit tests for validating data source based on pre and post-ecotone conditions
- adds a unit test to retrieve an actual calldata source from a pre-ecotone mainnet batcher tx

**Metadata**

- Fixes #162

### CODE REVIEW

**Comment by clabby** [comment]
Rebased on https://github.com/ethereum-optimism/kona/pull/186 - Closing & re-opening to bump CI.

File: crates/derive/testdata/raw_batcher_tx.hex
Comment (clabby) [inline crates/derive/testdata/raw_batcher_tx.hex:1]: Mind dropping a source for this tx?

Comment (merklefruit) [reply crates/derive/testdata/raw_batcher_tx.hex:1]: Sure! https://etherscan.io/tx/0xe69d94330faafb4f716f7ad9b3b50ea8ff5ce57aea6d2f8be07afb7fe49cd6cf

Comment (clabby) [reply crates/derive/testdata/raw_batcher_tx.hex:1]: Ty! Checks out

```sh
diff <(cat crates/derive/testdata/raw_batcher_tx.hex | xxd -p | tr -d '\n') <(cast rpc "debug_getRawTransaction" 0xe69d94330faafb4f716f7ad9b3b50ea8ff5ce57aea6d2f8be07afb7fe49cd6cf | jq -r | cut -c 3- | tr -d '\n')
```

**Pull request merged**